ARProgramming/ARProgramming/Controller/LevelSelectViewController.swift

```diff
@@ -57,29 +56,12 @@ class LevelSelectViewController: UIViewController {
     
 }
 
-// MARK: - UICollectionViewDataSource
-extension LevelSelectViewController: UICollectionViewDataSource {
-    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return levels.count
-    }
-    
-    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "levelCell", for: indexPath)
-        
-        if let levelCell = cell as? LevelCollectionViewCell {
-            let level = levels[indexPath.item]
-            levelCell.levelName.text = level.name
-            levelCell.unlocked = level.unlocked
-        }
-        
-        return cell
-    }
-}
-
 // MARK: - UICollectionViewDelegate
 extension LevelSelectViewController: UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        selectedLevel = levels[indexPath.item]
+        if let level  = collectionView.cellForItem(at: indexPath) as? LevelCollectionViewCell {
+            selectedLevel = level.level
+        }
         performSegue(withIdentifier: "arContainerSegue", sender: self)
```

Comment:
Maybe move _performSegue_ into the _if let ... {}_ such that we only perform a segue when a valid level has been selected.